backend/src/main/java/com/github/waymaker7/capstone/user/UserService.java

```diff
@@ -0,0 +1,34 @@
+package com.github.waymaker7.capstone.user;
+
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void createNewUser(UserCreationDTO userCreationDTO) {
+
+        if(!Objects.equals(userCreationDTO.getPassword(), userCreationDTO.getPasswordRepeat())){
+            throw new IllegalArgumentException("passwords do not match");
+        }
+
+        if(userCreationDTO.getUsername() == null || userCreationDTO.getUsername().isBlank()){
+            throw new IllegalArgumentException("username is blank");
+        }
+
+
```

Comment:
You could also check for the password fields to be empty and throw an exception in that case